python/paddle/amp/auto_cast.py

```diff
@@ -51,30 +93,33 @@ def amp_state():
 
 
 class AMPGlobalState:
-    def __init__(self):
+    def __init__(self) -> None:
         self.model_parameters = []
         self.use_master_grad = False
         self.already_register_final_backward_hook = False
         self.already_classify_params_meshes = False  # For dist
         self.mesh2params = {}  # For dist
         self.amp_dtype = 'float32'
 
-    def __setattr__(self, name, val):
+    def __setattr__(self, name: str, val: Any) -> dict[str, Any]:
         self.__dict__[name] = val
 
 
 _amp_global_state = AMPGlobalState()
 
 
-def amp_global_state():
+def amp_global_state() -> AMPGlobalState:
     return _amp_global_state
 
 
 # NOTE(zhiqiu): similar as paddle.static.amp.fp16_lists.AutoMixedPrecisionLists._update_list
 # The reason why not use AutoMixedPrecisionLists is that custom_black_varnames is not suitable for imperative mode.
 def _update_list(
-    custom_white_list, custom_black_list, level='O1', dtype='float16'
-):
+    custom_white_list: _CustomList,
+    custom_black_list: _CustomList,
+    level: AMP_LEVEL = 'O1',
+    dtype: dtype = 'float16',
```

Comment:
```suggestion
    dtype: str = 'float16',
```

这里是作为 `white_list()` 的 key，应该只能是 `str` ～